### DISCUSSION TITLE
Can't figure out why not null validation is being ignored in this case

### DISCUSSION THREAD

**Discussion by FatalCatharsis** [graphql-java/graphql-java#3072] [Q&A] [answered]
I'm trying to determine if I'm misunderstanding something about the function of graphql-java here. I have a schema with not null values, and I query a few of those not null fields. If I return an object from a data fetcher that does not have those properties, I would expect a validation error, but it omits them from the emitted content instead without validation errors. 

The schema for now is like this:

```
scalar Long

type Query {
	users : PaginatedSearchResult_User!
	user : User!
	organizations : PaginatedSearchResult_Organization!
}

type User {
	userId : Int!
	firstName : String!
	lastName : String!
	email : String!
	organizations : PaginatedSearchResult_Organization!
}

type Organization {
	orgId : Int!
	name : String!
}

type SearchError {
	error : String!
	errorCode : String!
	listErrors : [String!]!
}

type PaginatedType_Organization {
	totalCount : Int!
	edges : [PaginatedEdge_Organization!]!
	pageInfo : PageInfo!
}

type PaginatedEdge_Organization {
	nodes : [Organization!]!
	cursor : String!
}

type PageInfo {
	endCursor : String!
	hasNextPage : Boolean!
}

type PaginatedType_User {
	totalCount : Int!
	edges : [PaginatedEdge_User!]!
	pageInfo : PageInfo!
}

type PaginatedEdge_User {
	nodes : [User!]!
	cursor : String!
}

union PaginatedSearchResult_Organization = PaginatedType_Organization | SearchError

union PaginatedSearchResult_User = PaginatedType_User | SearchError
```

The query I tested was like this:
```
query UserQuery {
	users {
		__typename 
	
		... on PaginatedType_User {
			totalCount 
			edges {
				nodes {
					userId 
					firstName 
					lastName 
					organizations {
						__typename 
					
						... on PaginatedType_Organization {
							totalCount 
							edges {
								nodes {
									orgId 
									name 
								}
								cursor 
							}
							pageInfo {
								endCursor 
								hasNextPage 
							}
						}
						... on SearchError {
							error 
							errorCode 
							listErrors 
						}
					}
				}
				cursor 
			}
			pageInfo {
				endCursor 
				hasNextPage 
			}
		}
		... on SearchError {
			error 
			errorCode 
		}
	}
}
```
The outcome of the `ExecutionResult` has an empty list for `getErrors()` and `getData()` returns a hashmap with these values:

`{users={__typename=SearchError}}`

Shouldn't I have expected the execution to fail and a not null error present for the missing requested fields on `SearchError`?

**Comment by FatalCatharsis** [comment, marked as answer]
I've spent days trying to get to the bottom of what was happening, and I apologize, but not more than 10 minutes after posting this did I work out what was happening. I can't delete the post so I'll at least indicate what the problem was in case others follow in my footsteps, or an admin may delete this if they don't think it might be helpful. 

The issue was in the implementation of my `TypeResolver` for my union `PaginatedSearchResult_User `. I built my schema programmatically and while it correctly had the properties as not null, I was returning a new instance of `GraphQlObjectType` representing `SearchError` from the type resolver that didn't have not null.  I've since learned this is the incorrect way to do this, and I'm now looking up the existing object type on the schema provided from the `TypeResolutionEnvironment`.

**Reply by dondonz** [reply]
Hey @FatalCatharsis that's awesome that you solved your own question!
